target/linux/ath79/image/generic.mk

```diff
@@ -717,6 +721,28 @@ define Device/devolo_magic-2-wifi
 endef
 TARGET_DEVICES += devolo_magic-2-wifi
 
+define Device/dlink_covr-c1200-a1
+  SOC := qca9563
+  DEVICE_VENDOR := D-Link
+  DEVICE_MODEL := COVR-C1200
+  DEVICE_VARIANT := A1
+  DEVICE_PACKAGES := kmod-ath10k-ct ath10k-firmware-qca9888-ct
+  LOADER_TYPE := bin
+  LOADER_FLASH_OFFS := 0x060000
+  COMPILE := loader-$(1).bin loader-$(1).uImage
+  COMPILE/loader-$(1).bin := loader-okli-compile
+  COMPILE/loader-$(1).uImage := append-loader-okli $(1) | pad-to 64k | \
+	lzma | uImage lzma
+  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49
+  IMAGE_SIZE := 14528k
+  IMAGES += factory.bin
+  IMAGE/factory.bin := append-string hsqs | pad-to 64k | append-kernel | \
+    pad-to $$$$(BLOCKSIZE) | append-rootfs | pad-rootfs | check-size | \
+    pad-to 14876672 | append-file $(KDIR)/loader-$(1).uImage | \
+    pad-to 15990784 | dlink-sge-signature COVR-C1200 | dlink-sge-image
```

Comment:
you may be interested in the command
`append-squashfs-fakeroot-be`

and if big endian is wrong you can easily edit this to make the endianness an option
(I was looking into this, but then I found a way to not need it at all)

speaking of which, if the OEM upgrade process only verifies part of the magic number `hsqs` then it is now possible to set a custom kernel magic in both OKLI and DTS, and actually have the kernel there instead

also worth noting, the [lzma-loader has a small issue that's being resolved in another PR](https://github.com/openwrt/openwrt/pull/4143/commits/37b0a1364b9ad7a689aa9210996560a360e3a412)

also....I think it's preferred to use the `pad-to XXXk` syntax instead of just a number
so the number / 1024 and written with the 'k' represents the same value.